### DISCUSSION TITLE
How to read a model from a .lp format file

### DISCUSSION THREAD

**Discussion by liyix** [cvxpy/cvxpy#2286] [Q&A]
Hi everyone, I think I need help with how to read a problem model from a "model.lp" file? I currently use the solver (Cplex) to read the variables and constraints and convert them to cvxpy expressions. This approach is cumbersome and makes cvxpy take a long time to compile (especially at the "QpMatrixStuffing" step). Is there a better way?

**Comment by phschiele** [comment]
Hi @liyix, likely your approach ends up with an expression that is not vectorized, making the compilation slow.
CVXPY currently does not support reading a model from a `.lp` file, which describes a problem closer to standard form than you typically would in CVPXY. Thus, generating such files is rather supported by some solvers that CVXPY interfaces, including the CPLEX solver you are using.
If you need to convert a `.lp` file I would suggest trying to do so in a vectorized manner (i.e., read all variables first, and then try to create appropriately sized variables in CVXPY). It might be harder to do this generically.
